commit e611993564bf0bc200c430ef8d25699289c34c70
Author: Lars-Peter Clausen <lars@metafoo.de>
Date:   Tue Feb 3 00:16:19 2009 +0100

    struct serial_rs485 contains __32 fields and is exported to userspace so it
    needs linux/types.h in userspace aswell.

--- a/include/linux/serial.h
+++ b/include/linux/serial.h
@@ -10,8 +10,8 @@
 #ifndef _LINUX_SERIAL_H
 #define _LINUX_SERIAL_H
 
-#ifdef __KERNEL__
 #include <linux/types.h>
+#ifdef __KERNEL__
 #include <asm/page.h>
 
 /*
